### ISSUE TITLE
[question] Use of specific NCBI assemblies for bactopia --accessions

### ISSUE THREAD

**Issue by nmmahmed** [bactopia/bactopia#526]
Hello, I hope you are well! 

I'm able to download a specific NCBI Assembly via `bactopia --accession [accession]` -- is there a way to do so with a list of NCBI Assembly accessions (i.e bactopia --accessions [accessions_list.txt])? 

My understanding is that `bactopia search --accessions` works to prepare a list of Experiment accessions (associated with a given Study, Run, or BioSample), whereas my goal is to analyze a specific set of NCBI RefSeq assemblies. 

Thanks for your help!
Nashwa

**Comment by rpetit3** [comment]
hi @nmmahmed 

Let's see if we can get this fixed for you! Can you by chance share your accessions file or post the contents here?

Thank you!
Robert

**Comment by nmmahmed** [comment]
Hi Robert, 

Thanks for your response! I'm currently running bactopia locally with the following set of 10 (though I plan to analyze ~2000 including both NCBI and ENA assemblies): 

```
GCF_001857405.1
GCA_023167585.1
GCA_023167205.1
GCA_023165845.1
GCA_023164065.1
GCA_023164305.1
GCA_023164585.1
GCA_023163685.1
GCA_023162785.1
GCF_007624845.1
```

I intended to run `bactopia --accessions accessions.txt -profile arm --max_memory 64` with something like this: 

```
accession       runtype species genome_size
GCF_001857405.1 assembly_accession      Vibrio cholerae 4225000
GCA_023167585.1 assembly_accession      Vibrio cholerae 4225000
GCA_023167205.1 assembly_accession      Vibrio cholerae 4225000
GCA_023165845.1 assembly_accession      Vibrio cholerae 4225000
GCA_023164065.1 assembly_accession      Vibrio cholerae 4225000
GCA_023164305.1 assembly_accession      Vibrio cholerae 4225000
GCA_023164585.1 assembly_accession      Vibrio cholerae 4225000
GCA_023163685.1 assembly_accession      Vibrio cholerae 4225000
GCA_023162785.1 assembly_accession      Vibrio cholerae 4225000
GCF_007624845.1 assembly_accession      Vibrio cholerae 4225000
```

Thanks again!
Best,
Nashwa

**Comment by rpetit3** [comment]
I need to document this better, but could you try (added column header name)

```
accession
GCF_001857405.1
GCA_023167585.1
GCA_023167205.1
GCA_023165845.1
GCA_023164065.1
GCA_023164305.1
GCA_023164585.1
GCA_023163685.1
GCA_023162785.1
GCF_007624845.1
```

Then it would be `bactopia -profile ... --accessions acc.txt --genome_size 4225000 --species "Vibrio cholerae"`

I too think your second section should be directly supported, so I think I will get that added!

**Comment by nmmahmed** [comment]
With column header added: 
```
(bactopia) bash-5.2$ bactopia --accessions OneDrive\ -\ UW/bedford/cholera/bactopia/ncbi_acc.txt --outdir OneDrive\ -\ UW/bedford/cholera/bactopia/test -profile arm --max_cpus 4 --max_memory 64 --genome_size 4225000 --cleanup_workdir --max_time 300000 --species "Vibrio cholerae"
2024-06-14 11:52:49 INFO     2024-06-14 11:52:49:root:INFO - Checking if environment pre-builds are needed (this may take a while if building for the first time)                                      download.py:544

 N E X T F L O W   ~  version 24.04.2

Launching `/Users/nashwaahmed/micromamba/envs/bactopia/share/bactopia-3.0.1/main.nf` [nauseous_khorana] DSL2 - revision: 0cd9f79ba7



---------------------------------------------
   _                _              _             
  | |__   __ _  ___| |_ ___  _ __ (_) __ _       
  | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` |   
  | |_) | (_| | (__| || (_) | |_) | | (_| |      
  |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_| 
                            |_|                  
  bactopia v3.0.1
  Bactopia is a flexible pipeline for complete analysis of bacterial genomes 
---------------------------------------------
Core Nextflow options
  runName              : nauseous_khorana
  containerEngine      : docker
  container            : quay.io/bactopia/bactopia:3.0.1
  launchDir            : /Users/nashwaahmed
  workDir              : /Users/nashwaahmed/work
  projectDir           : /Users/nashwaahmed/micromamba/envs/bactopia/share/bactopia-3.0.1
  userName             : nashwaahmed
  profile              : arm
  configFiles          : /Users/nashwaahmed/micromamba/envs/bactopia/share/bactopia-3.0.1/nextflow.config

Required Parameters
  accessions           : OneDrive - UW/bedford/cholera/bactopia/ncbi_acc.txt

Dataset Parameters
  species              : Vibrio cholerae
  genome_size          : 4225000

QC Parameters
  adapters             : /Users/nashwaahmed/micromamba/envs/bactopia/share/bactopia-3.0.1/data/EMPTY_ADAPTERS
  phix                 : /Users/nashwaahmed/micromamba/envs/bactopia/share/bactopia-3.0.1/data/EMPTY_PHIX

Prokka Parameters
  proteins             : /Users/nashwaahmed/micromamba/envs/bactopia/share/bactopia-3.0.1/data/proteins.faa

Optional Parameters
  outdir               : OneDrive - UW/bedford/cholera/bactopia/test

Max Job Request Parameters
  max_memory           : 64
  max_time             : 300000

Nextflow Configuration Parameters
  cleanup_workdir      : true

Nextflow Profile Parameters
  condadir             : /Users/nashwaahmed/.bactopia/conda
  datasets_cache       : /Users/nashwaahmed/.bactopia/datasets
  singularity_cache_dir: /Users/nashwaahmed/.bactopia/singularity

!! Only displaying parameters that differ from the pipeline defaults !!
--------------------------------------------------------------------
If you use bactopia for your analysis please cite:

* Bactopia
  https://doi.org/10.1128/mSystems.00190-20

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://bactopia.github.io/acknowledgements/
--------------------------------------------------------------------
[-        ] process > BACTOPIA:DATASETS                        -
[-        ] process > BACTOPIA:GATHER:GATHER_MODULE            -
[-        ] process > BACTOPIA:GATHER:CSVTK_CONCAT             -
[-        ] process > BACTOPIA:QC:QC_MODULE                    -
[-        ] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE      -
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT          -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE        -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE         -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN -
[-        ] process > BACTOPIA:AMRFINDERPLUS:GENES_CONCAT      -
[-        ] process > BACTOPIA:AMRFINDERPLUS:PROTEINS_CONCAT   -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT               -
[-        ] process > BACTOPIA:DUMPSOFTWAREVERSIONS            -
ERROR ~ Cannot invoke method startsWith() on null object
[skipped  ] process > BACTOPIA:DATASETS                        [100%] 1 of 1, stored: 1 ✔
[-        ] process > BACTOPIA:GATHER:GATHER_MODULE            -
[-        ] process > BACTOPIA:GATHER:CSVTK_CONCAT             -
[-        ] process > BACTOPIA:QC:QC_MODULE                    -
[-        ] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE      -
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT          -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE        -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE         -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN -
[-        ] process > BACTOPIA:AMRFINDERPLUS:GENES_CONCAT      -
[-        ] process > BACTOPIA:AMRFINDERPLUS:PROTEINS_CONCAT   -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT               -
[-        ] process > BACTOPIA:DUMPSOFTWAREVERSIONS            -
ERROR ~ Cannot invoke method startsWith() on null object

 -- Check script 'micromamba/envs/bactopia/share/bactopia-3.0.1/./workflows/../lib/nf/bactopia.nf' at line: 171 or see '.nextflow.log' file for more details
```

**Comment by nmmahmed** [comment]
Can you recommend a `max_memory` and `max_time` parameter for this number of assemblies?

**Comment by rpetit3** [comment]
can you share the accession file here? I tried and it was able start processing, but ran into a different issue whish relared to the refseq accession being suppresed

See: https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_023164065.1/

**Comment by nmmahmed** [comment]
Hi, I hope you are well. 

I successfully ran bactopia with a modified accession file 

``` 
accession
GCF_001857405.1
GCF_007624845.1
```
using accessions that work individually (i.e. `bactopia --accession GCF_001857405.1`). However, there are accessions that do not appear to have issues, like [GCA_024015675.1](https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_024015675.1/) (no refseq suppression warning) but still result in errors that kill the entire run. Is there a way to have bactopia skip these rather than stopping? 

Example using GCA_024015675.1 which, when placed in the above accession list, throws the same error and kills the run: 

```
(bactopia) nahmed@gizmoj30:~/bac$ bactopia --accession GCA_024015675.1 --genome_size 4225000 --cleanup_workdir --species "Vibrio cholerae"
2024-06-26 14:42:09 INFO     2024-06-26 14:42:09:root:INFO - Checking if environment pre-builds are needed (this may take a while if building for the first time)                                  download.py:544

 N E X T F L O W   ~  version 24.04.2

Launching `/home/nahmed/miniconda3/envs/bactopia/share/bactopia-3.0.1/main.nf` [peaceful_shannon] DSL2 - revision: 0cd9f79ba7



---------------------------------------------
   _                _              _             
  | |__   __ _  ___| |_ ___  _ __ (_) __ _       
  | '_ \ / _` |/ __| __/ _ \| '_ \| |/ _` |   
  | |_) | (_| | (__| || (_) | |_) | | (_| |      
  |_.__/ \__,_|\___|\__\___/| .__/|_|\__,_| 
                            |_|                  
  bactopia v3.0.1
  Bactopia is a flexible pipeline for complete analysis of bacterial genomes 
---------------------------------------------
Core Nextflow options
  runName              : peaceful_shannon
  container            : quay.io/bactopia/bactopia:3.0.1
  launchDir            : /home/nahmed/bac
  workDir              : /home/nahmed/bac/work
  projectDir           : /home/nahmed/miniconda3/envs/bactopia/share/bactopia-3.0.1
  userName             : nahmed
  profile              : standard
  configFiles          : /home/nahmed/miniconda3/envs/bactopia/share/bactopia-3.0.1/nextflow.config, /home/nahmed/bac/nextflow.config

Required Parameters
  accession            : GCA_024015675.1

Dataset Parameters
  species              : Vibrio cholerae
  genome_size          : 4225000

QC Parameters
  adapters             : /home/nahmed/miniconda3/envs/bactopia/share/bactopia-3.0.1/data/EMPTY_ADAPTERS
  phix                 : /home/nahmed/miniconda3/envs/bactopia/share/bactopia-3.0.1/data/EMPTY_PHIX

Prokka Parameters
  proteins             : /home/nahmed/miniconda3/envs/bactopia/share/bactopia-3.0.1/data/proteins.faa

Nextflow Configuration Parameters
  cleanup_workdir      : true

Nextflow Profile Parameters
  condadir             : /home/nahmed/.bactopia/conda
  datasets_cache       : /home/nahmed/.bactopia/datasets
  singularity_cache_dir: /home/nahmed/.bactopia/singularity

!! Only displaying parameters that differ from the pipeline defaults !!
--------------------------------------------------------------------
If you use bactopia for your analysis please cite:

* Bactopia
  https://doi.org/10.1128/mSystems.00190-20

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://bactopia.github.io/acknowledgements/
--------------------------------------------------------------------
WARN: 
WARN: Conda Disclaimer
WARN: 
WARN: If you have access to Docker or Singularity, please consider
WARN: running Bactopia using containers. The containers are less
WARN: susceptible to Conda environment related issues (e.g. version
WARN: conflicts) and errors caused by creation of conda environments
WARN: in parallel (use '--max_cpus 1' to over come this error).
WARN: 
WARN: To use containers, you can use the profile parameter
WARN:     Docker: -profile docker
WARN:     Singularity: -profile singularity
WARN: 
--------------------------------------------------------------------

Each task will use 4 CPUs out of the available 16 CPUs. At most 
4 task(s) will be run at a time, this can affect the efficiency 
of Bactopia. You can use the '-qs' parameter to alter the number of 
tasks to run at a time (e.g. '-qs 2', means only 2 tasks or a maximum 
of 8 CPUs will be used at once)

executor >  local (4)
[skipped  ] process > BACTOPIA:DATASETS                             [100%] 1 of 1, stored: 1 ✔
[48/db714b] process > BACTOPIA:GATHER:GATHER_MODULE (GCA_024015675) [ 75%] 3 of 4, failed: 3, retries: 3
[-        ] process > BACTOPIA:GATHER:CSVTK_CONCAT                  -
[-        ] process > BACTOPIA:QC:QC_MODULE                         -
[-        ] process > BACTOPIA:ASSEMBLER:ASSEMBLER_MODULE           -
[-        ] process > BACTOPIA:ASSEMBLER:CSVTK_CONCAT               -
[-        ] process > BACTOPIA:SKETCHER:SKETCHER_MODULE             -
[-        ] process > BACTOPIA:ANNOTATOR:PROKKA_MODULE              -
[-        ] process > BACTOPIA:AMRFINDERPLUS:AMRFINDERPLUS_RUN      -
[-        ] process > BACTOPIA:AMRFINDERPLUS:GENES_CONCAT           -
[-        ] process > BACTOPIA:AMRFINDERPLUS:PROTEINS_CONCAT        -
[-        ] process > BACTOPIA:MLST:MLST_MODULE                     -
[-        ] process > BACTOPIA:MLST:CSVTK_CONCAT                    -
[-        ] process > BACTOPIA:DUMPSOFTWAREVERSIONS                 -
[33/cebd16] NOTE: Missing output file(s) `GCA_024015675-meta.tsv` expected by process `BACTOPIA:GATHER:GATHER_MODULE (GCA_024015675)` -- Execution is retried (1)
[e8/6fc2b6] NOTE: Missing output file(s) `GCA_024015675-meta.tsv` expected by process `BACTOPIA:GATHER:GATHER_MODULE (GCA_024015675)` -- Execution is retried (2)
[9d/376feb] NOTE: Missing output file(s) `GCA_024015675-meta.tsv` expected by process `BACTOPIA:GATHER:GATHER_MODULE (GCA_024015675)` -- Execution is retried (3)
ERROR ~ Error executing process > 'BACTOPIA:GATHER:GATHER_MODULE (GCA_024015675)'

Caused by:
  Missing output file(s) `GCA_024015675-meta.tsv` expected by process `BACTOPIA:GATHER:GATHER_MODULE (GCA_024015675)`


Command executed:

  MERGED="multiple-read-sets-merged.txt"
  mkdir -p fastqs
  mkdir -p extra
  
  if [ "assembly_accession" == "paired-end" ]; then
      # Paired-End Reads
      cp -L 001-r1 fastqs/GCA_024015675_R1.fastq.gz
      cp -L 001-r2 fastqs/GCA_024015675_R2.fastq.gz
      touch extra/empty.fna.gz
  elif [ "assembly_accession" == "single-end" ]; then
      # Single-End Reads
      cp -L 001-r1 fastqs/GCA_024015675.fastq.gz
      touch extra/empty.fna.gz
  elif [ "assembly_accession" == "ont" ]; then
      # Nanopore reads
      cp -L 001-r1 fastqs/GCA_024015675.fastq.gz
      touch extra/empty.fna.gz
  elif  [ "assembly_accession" == "hybrid" ] || [ "assembly_accession" == "short_polish" ]; then 
      # Paired-End Reads
      cp -L 001-r1 fastqs/GCA_024015675_R1.fastq.gz
      cp -L 001-r2 fastqs/GCA_024015675_R2.fastq.gz
      cp -L EMPTY_EXTRA extra/GCA_024015675.fastq.gz
  elif [ "assembly_accession" == "merge-pe" ] || [ "assembly_accession" == "hybrid-merge-pe" ] || [ "assembly_accession" == "short_polish-merge-pe" ]; then 
      # Merge Paired-End Reads
      echo "This sample had reads merged." > ${MERGED}
      echo "R1:" >> ${MERGED}
      find -name "*r1" | sort | xargs -I {} readlink {} | xargs -I {} ls -l {} | awk '{print $5"	"$9}' >> ${MERGED}
      find -name "*r1" | sort | xargs -I {} readlink {} | xargs -I {} cat {} > fastqs/GCA_024015675_R1.fastq.gz
      echo "Merged R1:" >> ${MERGED}
      ls -l fastqs/GCA_024015675_R1.fastq.gz | awk '{print $5"	"$9}' >> ${MERGED}
  
      echo "R2:" >> ${MERGED}
      find -name "*r2" | sort | xargs -I {} readlink {} | xargs -I {} ls -l {} | awk '{print $5"	"$9}' >> ${MERGED}
      find -name "*r2" | sort | xargs -I {} readlink {} | xargs -I {} cat {} > fastqs/GCA_024015675_R2.fastq.gz
      echo "Merged R2:" >> ${MERGED}
      ls -l fastqs/GCA_024015675_R2.fastq.gz | awk '{print $5"	"$9}' >> ${MERGED}
  
      if [ "assembly_accession" == "hybrid-merge-pe" ]; then
          cp -L EMPTY_EXTRA extra/GCA_024015675.fastq.gz
      else
          touch extra/empty.fna.gz
      fi
  elif [ "assembly_accession" == "merge-se" ]; then 
      # Merge Single-End Reads
      echo "This sample had reads merged." > ${MERGED}
      echo "SE:" >> ${MERGED}
      find -name "*r1" | sort | xargs -I {} readlink {} | xargs -I {} ls -l {} | awk '{print $5"	"$9}' >> ${MERGED}
      find -name "*r1" | sort | xargs -I {} readlink {} | xargs -I {} cat {} > fastqs/GCA_024015675.fastq.gz
      echo "Merged SE:" >> ${MERGED}
      ls -l fastqs/GCA_024015675.fastq.gz | awk '{print $5"	"$9}' >> ${MERGED}
  
      touch extra/empty.fna.gz
  elif [ "assembly_accession" == "sra_accession" ] || [ "assembly_accession" == "sra_accession_ont" ]; then
      if [ "4" == "3" ]; then
          echo "Unable to download GCA_024015675 from both SRA and ENA 3 times. This may or may 
              not be a temporary connection issue. Rather than stop the whole Bactopia run, 
              further analysis of GCA_024015675 will be discontinued." | \
          sed 's/^\s*//' > GCA_024015675-fastq-download-error.txt
          exit
      else
          # Download accession from ENA/SRA
          fastq-dl \
              --accession GCA_024015675 \
              --provider SRA \
              --cpus 4 \
              --outdir fastqs/ \
              --prefix GCA_024015675 \
              --group-by-experiment
          touch extra/empty.fna.gz
      fi 
  elif [ "true" == "true" ]; then
      if [ "assembly_accession" == "assembly_accession" ]; then
          if [ "4" == "3" ]; then
              touch extra/empty.fna.gz
              echo "Unable to download GCA_024015675 from NCBI Assembly 3 times. This may or may
                  not be a temporary connection issue. Rather than stop the whole Bactopia run, 
                  further analysis of GCA_024015675 will be discontinued." | \
              sed 's/^\s*//' > GCA_024015675-assembly-download-error.txt
              exit
          else
              # Verify Assembly accession
              check-assembly-accession.py GCA_024015675 > accession.txt 2> check-assembly-accession.txt
  
              if [ -s "accession.txt" ]; then
                  # Download from NCBI assembly and simulate reads
                  mkdir fasta/
                  ncbi-genome-download bacteria -o ./ -F fasta -p 4 \
                                              -u "https://ftp.ncbi.nlm.nih.gov/genomes" \
                                              -s genbank -A accession.txt -r 50 
                  find . -name "*GCA_024015675*.fna.gz" | xargs -I {} mv {} fasta/
                  rename 's/(GC[AF]_\d+).*/$1.fna.gz/' fasta/*
                  gzip -cd fasta/GCA_024015675.fna.gz > GCA_024015675-art.fna
                  rm check-assembly-accession.txt
              else
                  mv check-assembly-accession.txt GCA_024015675-assembly-accession-error.txt
                  exit
              fi
          fi
      elif [ "assembly_accession" == "assembly" ]; then
          if [ "false" == "true" ]; then
              gzip -cd EMPTY_EXTRA > GCA_024015675-art.fna
          else 
              cat EMPTY_EXTRA > GCA_024015675-art.fna
          fi
      fi
  
      # Simulate reads from assembly, reads are 250bp without errors
      art_illumina -p -ss MSv3 -l 250 -m 400 -s 30 --fcov 150 -ir 0 -ir2 0 -dr 0 -dr2 0 -rs 42                        -na -qL 33 -qU 40 -o GCA_024015675_R --id GCA_024015675 -i GCA_024015675-art.fna
  
      mv GCA_024015675_R1.fq fastqs/GCA_024015675_R1.fastq
      mv GCA_024015675_R2.fq fastqs/GCA_024015675_R2.fastq
      pigz -p 4 --fast fastqs/*.fastq
      cp GCA_024015675-art.fna extra/GCA_024015675.fna
      pigz -p 4 --best extra/GCA_024015675.fna
  fi
  
  # Validate input FASTQs
  IS_PAIRED="unknown"
  if [ "false" == "false" ]; then
      ERROR=0
      # Check paired-end reads have same read counts
      OPTS="--sample GCA_024015675 --min_basepairs 2241820 --min_reads 7472 --min_proportion 0.5 --runtype assembly_accession"
      if [ -f  "fastqs/GCA_024015675_R2.fastq.gz" ]; then
          # Paired-end
          IS_PAIRED="true"
          gzip -cd fastqs/GCA_024015675_R1.fastq.gz | fastq-scan > r1.json
          gzip -cd fastqs/GCA_024015675_R2.fastq.gz | fastq-scan > r2.json
          if ! reformat.sh in1=fastqs/GCA_024015675_R1.fastq.gz in2=fastqs/GCA_024015675_R2.fastq.gz qin=33 out=/dev/null 2> GCA_024015675-paired-end-error.txt; then
              ERROR=1
              echo "GCA_024015675 FASTQs contains an error. Please check the input FASTQs.
                  Further analysis is discontinued." | \
              sed 's/^\s*//' >> GCA_024015675-paired-end-error.txt
          else
              rm -f GCA_024015675-paired-end-error.txt
          fi
  
          if [[ -s r1.json ]] && [[ -s r2.json ]]; then
              if ! check-fastqs.py --fq1 r1.json --fq2 r2.json ${OPTS}; then
                  ERROR=1
              fi
          else
              NOT_GZIP=0
              if ! gzip -t fastqs/GCA_024015675_R1.fastq.gz; then
                  NOT_GZIP=1
              elif ! gzip -t fastqs/GCA_024015675_R2.fastq.gz; then
                  NOT_GZIP=1
              fi
  
              if [ "${NOT_GZIP}" -eq "0" ]; then
                  echo "GCA_024015675 FASTQs are empty. Please check the input FASTQs.
                      Further analysis is discontinued." | \
                  sed 's/^\s*//' > GCA_024015675-empty-error.txt
                  ERROR=1
              else
                  echo "GCA_024015675 FASTQs failed Gzip tests. Please check the input FASTQs.
                      Further analysis is discontinued." | \
                  sed 's/^\s*//' > GCA_024015675-gzip-error.txt
                  ERROR=1
              fi
          fi
          rm r1.json r2.json
      else
          # Single-end
          IS_PAIRED="false"
          gzip -cd fastqs/GCA_024015675.fastq.gz | fastq-scan > r1.json
  
          if [[ -s r1.json ]]; then
              if ! check-fastqs.py --fq1 r1.json ${OPTS}; then
                  ERROR=1
              fi
          else
              if ! gzip -t fastqs/GCA_024015675.fastq.gz; then
                  echo "GCA_024015675 FASTQs failed Gzip tests. Please check the input FASTQs.
                      Further analysis is discontinued." | \
                  sed 's/^\s*//' > GCA_024015675-gzip-error.txt
                  ERROR=1
              elif ! gzip -t fastqs/GCA_024015675_R2.fastq.gz; then
                  echo "GCA_024015675 FASTQs are empty. Please check the input FASTQs.
                      Further analysis is discontinued." | \
                  sed 's/^\s*//' > GCA_024015675-empty-error.txt
                  ERROR=1
              fi
          fi
          rm r1.json
      fi
  
      # Short polish should not be considered paired-end
      if [ "assembly_accession" == "short_polish" ]; then
          IS_PAIRED="false"
      fi
  
      # Failed validations so, let's keep them from continuing
      if [ "${ERROR}" -eq "1" ]; then
          mv fastqs/ failed-tests-fastqs/
      fi
  fi
  
  # Dump meta values to a TSV
  echo "sample<TAB>runtype<TAB>original_runtype<TAB>is_paired<TAB>is_compressed<TAB>species<TAB>genome_size" | sed 's/<TAB>/	/g' > GCA_024015675-meta.tsv
  echo "GCA_024015675<TAB>assembly_accession<TAB>assembly_accession<TAB>$IS_PAIRED<TAB>true<TAB>Vibrio cholerae<TAB>4225000" | sed 's/<TAB>/	/g' >> GCA_024015675-meta.tsv
  
  # Capture versions
  cat <<-END_VERSIONS > versions.yml
  "BACTOPIA:GATHER:GATHER_MODULE":
      art: $(echo $(art_illumina --help 2>&1) | sed 's/^.*Version //;s/ .*$//')
      fastq-dl: $(echo $(fastq-dl --version 2>&1) | sed 's/fastq-dl, version //')
      fastq-scan: $(echo $(fastq-scan -v 2>&1) | sed 's/fastq-scan //')
      ncbi-genome-download: $(echo $(ncbi-genome-download --version 2>&1))
      pigz: $(echo $(pigz --version 2>&1) | sed 's/pigz //')
  END_VERSIONS

Command exit status:
  0

Command output:
  (empty)

Work dir:
  /home/nahmed/bac/work/48/db714b5c923c6419cb19035d13d78c

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`

 -- Check '.nextflow.log' file for details
``` 

Thanks!

**Comment by MartinVad** [comment]
I’m not sure how the accessions file was created in your case or if this is still relevant for you, but we recently encountered an issue with accessions.txt files created using Windows PowerShell (named `accessions_windows.txt`). When we used this file as input for the `--accessions` option, we got the following error:

```java
ERROR ~ Cannot invoke method startsWith() on null object
``` 

After investigating, we found that the problem was caused by the line endings. The `accessions.txt` file created on Unix has LF line endings:

```console
$ file accessions.txt 
accessions.txt: ASCII text
```

Whereas the `accessions_windows.txt` file created on Windows has CRLF line endings:

```console
$ file accessions_windows.txt 
accessions_windows.txt: ASCII text, with CRLF line terminators
```

This difference in line endings was causing the error. Converting the line endings from CRLF to LF resolved the issue.
